src/web/common.rkt

```diff
@@ -8,6 +8,17 @@
   (match-define (list _ args expr) prog)
   (list 'FPCore args expr))
 
+(define (fpcore-add-props core props)
+  (match-define (list 'FPCore args expr) core)
+  `(FPCore ,args ,@props ,expr))
+
+(define (fpcore->string core)
+  (match-define (list 'FPCore args props ... expr) core)
+  (define props*  ; make sure each property (name, value) gets put on the same line
+    (for/list ([val (apply dict-set* '() props)]) ; how to make a list of pairs from a list
```

Comment:
Might as well make it `([(prop val) (in-dict (apply hash props))])`. That avoids the `car`/`cdr` later on.